cpp/ql/lib/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -295,10 +327,12 @@ private predicate fullBarrier(NodeEx node, Configuration config) {
     config.isBarrier(n)
     or
     config.isBarrierIn(n) and
-    not config.isSource(n)
+    not config.isSource(n) and
+    not config.isSource(n, _)
```

Comment:
Isn't this redundant? If `config.isSource(n)` is equivalent to `config.isSource(n, "")`, it should mean that if `not config.isSource(n, _)` holds, `not config.isSource(n, "")` must also hold.